index.js

```diff
@@ -2,53 +2,71 @@ var replace = require('gulp-replace');
 var path = require('path');
 var fs = require('fs');
 var checksum = require('checksum');
-var request = require('request');
 
 // Match {{cache-break:path/to/resource}}
 var reCacheBreak = /{{cache-break:(.+?)}}/g;
 
-// Match http://, https:// or // to indicate that the resource lies external
-// the the project.
-var reExternal = /^(?:https?:)?\/\//;
-
-var reLeadingSlash = /^\/+/;
-
-function cacheBreakPath(base, resource) {
-    var qs;
-    var time = Date.now();
-
-    // If we're not dealing with an external file and the file exists, let's
-    // get ourselves a checksum of the file's contents
-    if(!reExternal.test(resource)) {
-      // Since we know it's not external, we interpret the path as relative from
-      // the specified base.  This means that in order to determine the full path
-      // correclty we'll need to remove a single leading slash.
-      var fullPath = path.resolve(base, resource.replace(reLeadingSlash, ''));
-      if(fs.existsSync(fullPath)) {
-        qs = checksum(fs.readFileSync(fullPath));
-      }
-    }
+// Match {{cdn-path:path/to/resource}}
+var reCdnPath = /{{cdn-path:(.+?)}}/g;
 
-    // Fallback to the current timestamp
-    if(!qs) {
-      qs = time;
-    }
-
-    // Play well with an existing query-string
-    return resource + (resource.indexOf('?') === -1 ? '?' : '&') + 'cb=' + qs;
+function CacheBreaker() {
+  this.checksumCache = {};
 }
 
-module.exports = function(base) {
-  base = base && path.resolve(base);
+CacheBreaker.prototype.cacheBreakPath = function(base, resource) {
+  base = base || process.cwd();;
+
+  var joinedPath = path.join(base, resource);
+  var fullPath = path.resolve(joinedPath);
 
-  if(!base) {
-    base = process.cwd();
+  var cs;
+  var mtime = fs.statSync(fullPath).mtime.getTime();
+  if (fullPath in this.checksumCache && mtime === this.checksumCache[fullPath].mtime) {
+    cs = this.checksumCache[fullPath].checksum;
+  } else {
+    cs = checksum(fs.readFileSync(fullPath));
+    this.checksumCache[fullPath] = { checksum: cs, mtime: mtime };
   }
 
+  var dirname = path.dirname(resource);
+  var extname = path.extname(resource);
+  var basename = path.basename(resource, extname);
+
+  return path.join(dirname, basename + '.' + cs.substring(0, 10) + extname);
+};
+
+CacheBreaker.prototype.cdnUri = function(base, resource, host) {
+  if (host) {
+    return 'https://' + host + this.cacheBreakPath(base, resource);
```

Comment:
Should likely make this configurable, just in case they're using `http`.  Maybe a third param, `secure`?